src/artifacts.ml

```diff
@@ -1,52 +1,21 @@
 open! Stdune
 open Import
-open Dune_file
 
 type t =
   { context     : Context.t
-  ; local_bins  : Path.t String.Map.t
+  ; local_bins  : Path.t String.Map.t lazy_t
   ; public_libs : Lib.DB.t
   }
 
-let create (context : Context.t) ~public_libs l ~f =
+let create (context : Context.t) ~public_libs ~build_system =
   let bin_dir = Config.local_install_bin_dir ~context:context.name in
   let local_bins =
-    List.fold_left l ~init:String.Map.empty ~f:(fun acc x ->
-      List.fold_left (f x) ~init:acc ~f:(fun local_bins stanza ->
-        match (stanza : Stanza.t) with
-        | Install { section = Bin; files; _ } ->
-          List.fold_left files ~init:local_bins
-            ~f:(fun acc { Install_conf. src; dst } ->
-              let name =
-                match dst with
-                | Some s -> s
-                | None -> Filename.basename src
-              in
-              let key =
-                if Sys.win32 then
-                  Option.value ~default:name
-                    (String.drop_suffix name ~suffix:".exe")
-                else
-                  name
-              in
-              let in_bin_dir =
-                let fn =
-                  if Sys.win32 then
-                    match Filename.extension src with
-                    | ".exe" | ".bc" ->
-                      if Filename.extension name <> ".exe" then
-                        name ^ ".exe"
-                      else
-                        name
-                    | _ -> name
-                  else
-                    name
-                in
-                Path.relative bin_dir fn
-              in
-              String.Map.add acc key in_bin_dir)
-        | _ ->
-          local_bins))
+  lazy (
+    Build_system.targets_of build_system ~dir:bin_dir
+    |> Path.Set.fold ~init:String.Map.empty ~f:(fun path acc ->
+      let name = Filename.basename (Path.to_string path) in
```

Comment:
If you keep the `let key = if Sys.win32 ... in` and use `key` instead of `name` in the following line, then the Windows error should go away. The idea is that the key in the map are the executable names without the `.exe`, even on Windows.

Reply:
Thanks. This regression seems fixed now!